### ISSUE TITLE
python_requirements environment constraints

### ISSUE THREAD

**Issue by bvolkmer** [replicate/cog#2078]
If I have or example, a requirements.txt with the following content:
```
pywin32==308 ; platform_system == 'Windows'
```
It will try to install the package and fail because it is not available on Linux.
But normally, pip (at least 24.3.1) does succeed with
```
Ignoring pywin32: markers 'platform_system == "Windows"' don't match your environment
```
So I suspect that the requirement.txt file is modified in the process.

**Comment by meatballhat** [comment]
@bvolkmer You are absolutely right. Although the original `requirements.txt` is not modified, there is [a decent amount of fiddling going on](https://github.com/replicate/cog/blob/4161a39215004f98fe9284b99cb6600d9b67d22e/pkg/config/config.go#L327). If it's an option, I'd recommend using a separate file specifically for the Cog environment such as `python_requirement: requirements-cog.txt` which excludes the non-Linux dependencies.

**Comment by anish-sahoo** [comment]
Closing due to inactivity. If this is still a concern, please reopen the issue or open a new issue.
